### ISSUE TITLE
Feature Request: Select other layouts in effect editors

### ISSUE THREAD

**Issue by SamantazFox** [polychromatic/polychromatic#373]
Could you please propose (or allow to create) other keyboard layouts for the effects editor?
That would be a nice addition :)

**Comment by lah7** [comment]
It's possible already by choosing **Edit Metadata** and picking another graphic mapping (or grid),  but only if the other device has the same dimensions (rows/columns). If you mean to create a new graphic for a new device or locale, [see the documentation on device maps](https://docs.polychromatic.app/devicemaps/).

Or, did you mean, being able to create effects for a different device you might not have? Like _"I have a BlackWidow Chroma, but I'd like to make an effect for my friend's Razer Blade"_ use case?

**Comment by SamantazFox** [comment]
> If you mean to create a new graphic for a new device or locale, [see the documentation on device maps](https://docs.polychromatic.app/devicemaps/).

Yeah, it's what I meant. Though, my original idea was to have different layouts (AZERTY, QWERTZ, etc...) on the same graphic, but from that link I realized that the layout is part of the graphic. If I have time, I'll make a layout generator...

**Comment by lah7** [comment]
It would make sense to me to separate the key labels from the layout, since the current mechanism only encourages duplicate graphics with small changes, which will become a maintenance burden. 🛠️  I have some notes on improving the metadata _(key codes, mapping LED positions to labels)_ so abstracting labels from physical layout seems like something I will do as part of those changes.

The key labels in the SVG are `<text>`, so changing them is pretty simple for the computer, if just a bit tedious.

💡 I have some ideas to improve this area, thanks for flagging it up!

**Comment by SamantazFox** [comment]
> The key labels in the SVG are `<text>`, so changing them is pretty simple for the computer, but just a bit tedious for us humans. 😄

Could it be possible to generate the SVG on the fly? Something like different patterns for the keys (some layouts place the symbols differently, see screenshots below), which are alway the same size (a square). The base SVG would only represent the keyboard with un-labelled keys (except a few ones like space, tab or other layout-agnostic keys), and there would be a metadata file along that SVG which lists where are the keys (i.e where the previously generated "square" has to go), with some sort of ID to easily map them. Then the only thing left to do is to provide what's on the different keys (i.e `key_R1-1 = ²`).

AZERTY vs QWERTY:
![image](https://user-images.githubusercontent.com/52980486/141391512-c44fbfc2-fbd1-41ed-be9d-dca4d6886b7c.png) ![image](https://user-images.githubusercontent.com/52980486/141391570-25f0bedf-d02a-435b-9d38-ed9a4ca8db65.png)

(full size: [AZERTY](https://qcqcdn.com/euroclintrials/product/3033628887-8886419345602.jpg) and [QWERTY](https://www.sferaufficio.com/immagini/http://inishop.com/img/gallery/36382040_3307662627.jpg))

**Comment by lah7** [comment]
Possible, but generating the entire SVG would be impractical since there are non-keyboards too. I did some brainstorming, and this is my plan:
```
- Each graphic becomes a physical layout: ANSI (US) / ISO (Intl) with QWERTY as default
    blackwidow_chroma_m_keys_ANSI.svg   (US)
    blackwidow_chroma_m_keys_ISO.svg    (GB)
    mamba.svg                           (Agnostic)

    New class to be added for each key: "sc-22" (or "scancode-22"?) alongside ID x0-y0 for LED positioning

    Ref: https://en.wikipedia.org/wiki/File:Physical_keyboard_layouts_comparison_ANSI_ISO_KS_ABNT_JIS.png

- Each graphic has a JSON file, superseding existing JSON files.

    {
        "cols": 22,
        "rows": 6,
        "key_labels": true,       // When false, no further changes made
        "scancode_to_matrix": {
            "22": [2,3],
            "23": [3,3],
            ...
        }
    }

- Need to automatically map keyboard_layout (en_GB, de_DE, etc) to correct physical layout / arrangement,
  and/or allow user to choose (in "Device Info"?)

    {
        "en_GB": {
            "physical": "ISO",
            "layout": "QWERTY_UK"
            "inherits": "QWERTY_US"
        },
        "en_US": {
            "physical": "ISO",
            "layout": "QWERTY_US"
            "inherits": null
        }
    }

    Could disable key labels by default for layouts not yet implemented?

- A new "layouts/QWERTY_US.json" maps scan codes to labels.  (#373)
    {
        "21": "Y",
        ...
    }

    e.g. layouts/QWERTZ.json:
    {
        "21": "Z",
        ...
    }

    The editor will change the <text> key labels, and re-align them horiziontally.

- Each graphic JSON can override the key label if styling is different
    {
        "overrides": {
            "28": {
                "QWERTY": "enTer",
            }
        }
    }

- New tests to validate SVG/JSON integrity
- Nice to have: Dev GUI to edit keyboard metadata (map scan codes to QWERTY/QWERTZ/AZERTY).
- Nice to have: GUI to test scan code mappings (use Qt while in focus?)
```

For the layouts that use symbols used in place of text, (like a QWERTY <kbd>Caps LK</kbd> == AZERTY <kbd>🔒</kbd>) there should be a Unicode text character to use. (On here, they're turning into emojis)
